src/legacy/core_plugins/ui_metric/server/usage/collector.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const UI_METRIC_USAGE_TYPE = 'ui_metric';
+
+export function registerUiMetricUsageCollector(server: any) {
+  const collector = server.usage.collectorSet.makeUsageCollector({
+    type: UI_METRIC_USAGE_TYPE,
+    fetch: async (callCluster: any) => {
+      const { SavedObjectsClient, getSavedObjectsRepository } = server.savedObjects;
```

Comment:
I was wondering if it makes sense to wrap everything in this function in a `try..catch`, but it seems like the [downstream usage](https://github.com/elastic/kibana/blob/master/src/legacy/server/usage/classes/collector_set.js#L87) already has error handling if I am following the logic correctly

Reply:
Really good idea, I did not realize the collector implementation was catching and surfacing errors. I think you're right that it isn't needed here since it's handled downstream.